preview/MsixCore/MsixCoreInstaller/RegistryDevirtualizer.cpp

```diff
@@ -68,6 +68,54 @@ HRESULT RegistryDevirtualizer::Run(_In_ bool remove)
     return S_OK;
 }
 
+HRESULT RegistryDevirtualizer::DeleteKeyIfPresent(HKEY rootHive, std::wstring subKeyPath, std::wstring extensionName)
+{
+    if (!m_hiveFileNameExists)
+    {
+        return S_OK;
+    }
+    std::wstring rootPath = m_loadedHiveKeyName + L"\\Registry";
+    RETURN_IF_FAILED(m_rootKey.Open(HKEY_USERS, rootPath.c_str(), KEY_READ)); // root key until registry
+
+    // map real root hive to virtual root hive
+    PCWSTR virtualKeyPath = L"";
+    for (auto mapping : mappings)
+    {
+        if (mapping.actualRootHive == rootHive && mapping.actualSubkey == subKeyPath)
+        {
+            virtualKeyPath = mapping.virtualKey;
+        }
+    }
+
+    // Per user mapping
+    RegistryKey userClassesKey;
+    HRESULT hrOpenUserClassesKey = m_rootKey.OpenSubKey(virtualKeyPath, KEY_READ, &userClassesKey);
+    if (SUCCEEDED(hrOpenUserClassesKey))
+    {
+        RegistryKey extensionKey;
+        HRESULT hrUserExtensionKey = userClassesKey.OpenSubKey(extensionName.c_str(), KEY_READ, &extensionKey);

```

Comment:
>KEY_READ [](start = 86, length = 8)

test this out, but i think you need KEY_WRITE in order to delete.

Reply:
Updated to KEY_WRITE. Tested that key gets deleted with this in the debugger.